packages/devtools_app/lib/src/inspector/inspector_controller.dart

```diff
@@ -581,42 +580,56 @@ class InspectorController extends DisposableController
     _selectionGroups.cancelNext();
 
     final group = _selectionGroups.next;
-    final pendingSelectionFuture = group.getSelection(
-      selectedDiagnostic,
-      treeType,
-      isSummaryTree: isSummaryTree,
-    );
 
-    final Future<RemoteDiagnosticsNode> pendingDetailsFuture = isSummaryTree
-        ? group.getSelection(selectedDiagnostic, treeType, isSummaryTree: false)
-        : null;
+    if (firstFrame) {
+      final queryParams = loadQueryParams();
+      if (queryParams.containsKey(inspectorRefQueryParam)) {
+        print('trying to select from service');
+        await group.setSelectionInspector(
+          InspectorInstanceRef(queryParams[inspectorRefQueryParam]),
+          false,
+        );
+      }
+    } else {
```

Comment:
this shouldn't be an else. You should read the selection from the flutter device after you set the selection on the device. That will make everything work without you needing to duplicate logic for separating out the details and summary selections, etc.

Reply:
done